cdap-explore-client/src/main/java/co/cask/cdap/explore/service/ExploreServiceManager.java

```diff
@@ -42,7 +46,12 @@ public boolean isServiceEnabled() {
 
   @Override
   public int getMaxInstances() {
-    return cConf.getInt(Constants.Explore.MAX_INSTANCES, 1);
+    try {
+      int configuredInstances = cConf.getInt(Constants.Explore.CONTAINER_INSTANCES);
```

Comment:
I think you should use `cConf.get()`, and simply do the warning if its non null.
That way, you don't need to catch any NPE.